sklego/ensemble_model.py

```diff
@@ -0,0 +1,104 @@
+from sklearn.base import clone
+from sklearn.utils.validation import check_is_fitted
+from sklearn.metrics import confusion_matrix
+
+import numpy as np
+
+class EnsembleModel():
+    """
+    A classifier that takes the uncertainty p(class | model) of a classifier into account.
+    Based on the requirements of issue #23 of koaning/scikit-lego
+    """
+
+    def __init__(self, estimator=None, alpha=0.5, X_test=None, y_test=None):
+        """
+
+        :param estimator: sklean classifier; (eg. LogisticRegression, Decision Tree etc.)
+        :param alpha: hyperparameter; needs to be optimized to increase classification accuracy
+        :param X_test: array of feature values that will be used to test the classifier and create a confusion matrix
+        :param y_test: array of target values that will be used to test the classifier and create a confusion matrix
+        """
+        self.estimator = estimator
+        self.alpha = alpha
+        self.X_test = X_test
+        self.y_test = y_test
+
+    def fit(self, X, y):
+        """
+
+       :param X: array of feature values used for training the classifier
+       :param y: array of target values used for training the classifier
+       :return:
+                - trained classifier (self.orig_estimator)
+                - confusion matrix populated with the probabilities of each real value vs model prediction value
+                  of the target variable (self.confusion_matrix)
+                - updated prediction probabilities of each target value made by the trained classifier
+                  (self.prob_matrix)
+       """
+        self.estimator_ = clone(self.estimator)
+        self.orig_estimator_ = self.estimator_.fit(X, y)
```

Comment:
You now have `self.estimator_` and `self.orig_estimator_` referring to the same estimator, I would get rid of one of them